rbac/management/group/view.py

```diff
@@ -771,6 +785,27 @@ def principals(self, request: Request, uuid: Optional[UUID] = None):
             # Serialize the group...
             output = GroupSerializer(group)
             response = Response(status=status.HTTP_200_OK, data=output.data)
+
+            if status.is_success(response.status_code):
+                if len(principals) > 0:
```

Comment:
this should be part of transaction block which start on 715.
you can put audit log creation under those related conditions (`len(service_accounts|principals) > 0`).

if operation fails, transaction is rolled back and record into audit log is not created.